### PR TITLE
feat: button chooser rte plugin

### PR DESCRIPTION
**Pull request by mike-schmid**

### CODE REVIEW

File: src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useFloatingButtonInsert.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    });
+
+    // wait for update before focusing input
+    useEffect(() => {
+        if (open) {
+            update();
+            floatingButtonActions.updated(true);
+        } else {
+            floatingButtonActions.updated(false);
+        }
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useFloatingButtonInsert.ts:69]:
```suggestion
        if (open) {
            update();
        }
        floatingButtonActions.updated(open);
```

File: src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useVirtualFloatingButton.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { UseVirtualFloatingOptions, flip, offset, useVirtualFloating } from '@udecode/plate-floating';
+
+export const useVirtualFloatingButton = (floatingOptions?: UseVirtualFloatingOptions) => {
+    return useVirtualFloating({
+        placement: 'bottom-start',
+        middleware: [
+            offset(12),
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useVirtualFloatingButton.ts:7]: set as const

File: src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useVirtualFloatingButton.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+            flip({
+                padding: 96,
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useVirtualFloatingButton.ts:9]: set as const

File: src/components/RichTextEditor/Plugins/ButtonPlugin/components/index.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/**
+ * @file Automatically generated by barrelsby.
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ButtonPlugin/components/index.ts:2]: remove

File: src/components/RichTextEditor/Plugins/ButtonPlugin/index.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @file Automatically generated by barrelsby.
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ButtonPlugin/index.ts:2]: remove

File: src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/insertButton.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { InsertNodesOptions, PlateEditor, TText, Value, insertNodes } from '@udecode/plate-core';
+import { TButtonElement } from '../types';
+import { CreateButtonNodeOptions, createButtonNode } from '../utils/index';
+
+/**
+ * Insert a button node.
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/insertButton.ts:6]: remove comment or rename function if necessary

File: src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/submitFloatingButton.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import { PlateEditor, Value, focusEditor, getPluginOptions } from '@udecode/plate-core';
+import { floatingButtonActions, floatingButtonSelectors } from '../components/FloatingButton/floatingButtonStore';
+import { ButtonPlugin, ELEMENT_BUTTON } from '../createButtonPlugin';
+import { upsertButton } from './index';
+
+/**
+ * Insert button if url is valid.
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/submitFloatingButton.ts:7]: could you split it up in functions with the according naming and remove the comments?

File: src/components/RichTextEditor/Plugins/ButtonPlugin/withButton.ts [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+    withoutNormalizing,
+} from '@udecode/plate-core';
+import { withRemoveEmptyNodes } from '@udecode/plate-normalizers';
+import { Path, Point, Range } from 'slate';
+import { upsertButton } from './transforms/index';
+import { ButtonPlugin, ELEMENT_BUTTON } from './createButtonPlugin';
+import { AnyObject } from '@udecode/plate';
+
+/**
+ * Insert space after a url to wrap a bztton.
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ButtonPlugin/withButton.ts:29]:
```suggestion
 * Insert space after a url to wrap a button.
```

File: src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/index.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+/**
```
[the remark this answers is not in the document]
Comment (mike-schmid) [reply src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/index.ts:1]: done

File: src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useFloatingButtonInsert.ts (hunk 1 above)
Comment (mike-schmid) [reply src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useFloatingButtonInsert.ts:69]: done

File: src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useVirtualFloatingButton.ts (hunk 2 above)
Comment (mike-schmid) [reply src/components/RichTextEditor/Plugins/ButtonPlugin/components/FloatingButton/useVirtualFloatingButton.ts:7]: done

File: src/components/RichTextEditor/Plugins/ButtonPlugin/components/index.ts (hunk 4 above)
Comment (mike-schmid) [reply src/components/RichTextEditor/Plugins/ButtonPlugin/components/index.ts:2]: done

File: src/components/RichTextEditor/Plugins/ButtonPlugin/index.ts (hunk 5 above)
Comment (mike-schmid) [reply src/components/RichTextEditor/Plugins/ButtonPlugin/index.ts:2]: done

File: src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/insertButton.ts (hunk 6 above)
Comment (mike-schmid) [reply src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/insertButton.ts:6]: done

File: src/components/RichTextEditor/Plugins/ButtonPlugin/withButton.ts (hunk 8 above)
Comment (mike-schmid) [reply src/components/RichTextEditor/Plugins/ButtonPlugin/withButton.ts:29]: done

File: src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/submitFloatingButton.ts (hunk 7 above)
Comment (mike-schmid) [reply src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/submitFloatingButton.ts:7]: done. I removed the comment. It's clear what the method does anyway....

**Review by triggertoo — APPROVED** [review]

**Pull request merged**